test: Build centos-7 from latest image

dperpeet opened the pull request:
This way we don't miss an update but get it automatically when it's
available. We explicitly don't want to change the major version
automatically at this point.

```shell
[admin@localhost ~]$ CENTOS_LATEST=$(virt-builder -l | sort -r | grep -m 1 centos-7 | cut -d" " -f1)
[admin@localhost ~]$ echo "$CENTOS_LATEST"
centos-7.3
```

dperpeet commented:
testing image creation

dperpeet commented:
bot: Image refresh for centos-7

cockpituous commented:
Image creation for centos-7 in process on cockpit-8.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-centos-7-8d61ea11ee276c60866cfed28c28ef1e7b1dc468/

dperpeet commented:
Looks good:
```
++ dirname /build/cockpit/test/images/scripts/virt-builder-fedora
+ BASE=/build/cockpit/test/images/scripts
+ out=/build/cockpit/test/tmp/run/centos-7.qcow2
+ os=centos-7.3
```

cockpituous commented:
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-centos-7-8d61ea11ee276c60866cfed28c28ef1e7b1dc468

martinpitt commented on test/images/scripts/centos-7.bootstrap:
```diff
@@ -1,4 +1,7 @@
 BASE=$(dirname $0)
-$BASE/virt-builder-fedora "$1" centos-7.3 x86_64
+# get the latest centos-7 virt-builder has to offer
+CENTOS_LATEST=$(virt-builder -l | sort -r | grep -m 1 centos-7 | cut -d" " -f1)
```
I think you should extend the grep for `centos-7.*x86_64` to avoid catching e. g. the `aarch64` or any other architecture image, just in case that gets available earlier.

martinpitt commented on test/images/scripts/centos-7.bootstrap:
```diff
@@ -1,4 +1,7 @@
+# get the latest centos-7 virt-builder has to offer
+CENTOS_LATEST=$(virt-builder -l | sort -r | grep -m 1 centos-7 | cut -d" " -f1)
+
+$BASE/virt-builder-fedora "$1" "$CENTOS_LATEST" x86_64
```
To avoid duplicating the architecture, you could even do
```sh
CENTOS_LATEST=$(virt-builder -l | sort -r | grep -m 1 -o 'centos-7.*x86_64')
$BASE/virt-builder-fedora "$1" $CENTOS_LATEST
```

(no quotes around the variable as it will contain the name *and* the arch and they are two arguments).

dperpeet commented:
> I suppose this is (hopefully) a theoretical error only, but I think let's keep the current x86_64 selection.

Thanks, I changed this and moved the interesting parts into variables in case we want to copy/paste this in the future.

test:
```shell
DISTRO=centos-7
[admin@localhost ~]$ ARCH=x86_64
[admin@localhost ~]$ CENTOS_LATEST=$(virt-builder -l | grep "$ARCH" | sort -r | grep -m 1 "$DISTRO" | cut -d" " -f1)
[admin@localhost ~]$ echo "'$CENTOS_LATEST'"
'centos-7.3'
```

dperpeet commented:
I can confirm that `virt-builder` ignores `--arch` when using `-l` to list.

martinpitt commented:
I verified that this picks the "centos-7.3" image, so IMHO it's not necessary to actually rebuild it.
